packages/mdc-toolbar/foundation.js

```diff
@@ -0,0 +1,234 @@
+      getViewportScrollY: () => /* number */ 0,
+      getOffsetHeight: () => /* number */ 0,
+      getFlexibleRowElementOffsetHeight: () => /* number */ 0,
+      notifyChange: (/* evtData: {detail: number} */) => {},
+      setStyleForRootElement: (/* property: string, value: string */) => {},
+      setStyleForTitleElement: (/* property: string, value: string */) => {},
+      setStyleForFlexibleRowElement: (/* property: string, value: string */) => {},
+      setStyleForFixedAdjustElement: (/* property: string, value: string */) => {},
+    };
+  }
+
+  constructor(adapter) {
+    super(Object.assign(MDCToolbarFoundation.defaultAdapter, adapter));
+    this.resizeHandler_ = () => this.setKeyHeights();
+    this.scrollHandler_ = () => this.changeToolbarStyles();
+    this.resizeFrame_ = 0;
+    this.scrollFrame_ = 0;
+    this.executedLastChange_ = false;
+
+    // Initialize toolbar behavior configuaration variables
+    this.keyHeights = {
+      'toolbarRowHeight': 0,
+      // Calculated Height ratio. We use ratio to calculate corresponding heights in resize event.
+      'toolbarRatio': 0,  // The ratio of toolbar height to row height
+      'flexibleExpansionRatio': 0, // The ratio of flexible space height to row height
+      'maxTranslateYRatio': 0, // The ratio of max toolbar move up distance to row height
+      'scrollThesholdRatio': 0, // The ratio of max scrollTop that we should listen to to row height
+      // Derived Heights based on the above key ratios.
+      'toolbarHeight': 0,
+      'flexibleExpansionHeight': 0, // Flexible row minus toolbar height (derived)
+      'maxTranslateYDistance': 0, // When toolbar only fix last row (derived)
+      'scrollTheshold': 0,
+    };
+    // Toolbar fixed behavior
+    // If fixed is targeted only at the last row
+    this.fixedLastrow_ = false;
+    // If toolbar is fixed
+    this.fixedAllRow_ = false;
+    // Toolbar flexible behavior
+    // If the first row is flexible
+    this.hasFlexibleRow_ = false;
+    // If use the default behavior
+    this.useFlexDefaultBehavior_ = false;
+  }
+
+  init() {
+    this.fixedAllRow_ = this.adapter_.hasClass(MDCToolbarFoundation.cssClasses.FIXED);
+    this.fixedLastrow_ = this.adapter_.hasClass(MDCToolbarFoundation.cssClasses.FIXED_LASTROW);
+    this.hasFlexibleRow_ = this.adapter_.hasClass(MDCToolbarFoundation.cssClasses.TOOLBAR_ROW_FLEXIBLE);
+    if (this.hasFlexibleRow_) {
+      this.useFlexDefaultBehavior_ = this.adapter_.hasClass(MDCToolbarFoundation.cssClasses.USE_FLEX_DEFAULT_BEHAVIOR);
+    }
+    this.initKeyRatio_();
+    this.setKeyHeights_({isInit: true});
+    this.adapter_.addClass(MDCToolbarFoundation.cssClasses.ROOT);
+    this.adapter_.registerResizeHandler(this.resizeHandler_);
+    this.adapter_.registerScrollHandler(this.scrollHandler_);
+  }
+
+  destroy() {
+    this.adapter_.deregisterResizeHandler(this.resizeHandler_);
+    this.adapter_.deregisterScrollHandler(this.scrollHandler_);
+  }
+
+  setKeyHeights() {
```

Comment:
`setKeyHeights` and `updateToolbarStyles` make sense to make them private because they are only called by `resize` and `scroll` handler. But one question would be if we make them all private, how would be properly test the behaviors?